catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/FrameworkProperties.java

```diff
@@ -215,11 +215,11 @@ public void setConnectedSources(List<ConnectedSource> connectedSources) {
     this.connectedSources = connectedSources;
   }
 
-  public Map<String, FederatedSource> getFederatedSources() {
+  public Collection<FederatedSource> getFederatedSources() {
     return federatedSources;
   }
 
-  public void setFederatedSources(Map<String, FederatedSource> federatedSources) {
+  public void setFederatedSources(Collection<FederatedSource> federatedSources) {
```

Comment:
❓ Should we make a defensive copy here?

Reply:
Shouldn't need to, blueprint should inject this as a [Dynamic Collection](https://github.com/apache/aries/blob/trunk/blueprint/blueprint-core/src/main/java/org/apache/aries/blueprint/utils/DynamicCollection.java) that takes care of concurrency control for us.

Reply:
As for making it non modifiable that's a good question. I don't think we'd need to as nothing else in the framework properties does that